### ISSUE TITLE
In some regions, monday is the first day of the week

### ISSUE THREAD

**Issue by mrvanes** [go-gitea/gitea#6058]
Gitea version: 1.7.1, from single binary.
The high-level contribution overview starts the week at sunday. I'd prefer to make it start on monday and have sunday the bottom, as a feature-request.
![image](https://user-images.githubusercontent.com/1901782/52721242-8c387700-2fa9-11e9-9950-8a09fbdc06b6.png)

**Comment by kolaente** [comment]
I'd prefer to have this configurable when it gets implemented as the week starts in some countries on monday while in others on sunday.

**Comment by jolheiser** [comment]
From me tinkering with this for a bit:
  
While the module we use allows us to change the order of the days of the week, the squares do not seem to change accordingly to match, which unfortunately means this feature may have to first happen upstream...

**Comment by Oreolek** [comment]
So this depends on WildCodeSchool/vue-calendar-heatmap#1 right?

**Comment by kolaente** [comment]
@Oreolek Yeah pretty much.

**Comment by MalcolmAnderson** [comment]
Is any work being done on this? +1

**Comment by mind-overflow** [comment]
any update?

**Comment by lafriks** [comment]
This depends on upstream library to add such option so this should be asked in it

**Comment by noerw** [comment]
upstream reference with a PR: https://github.com/julienr114/vue-calendar-heatmap/pull/14

**Comment by molaie** [comment]
Any chance for this feature? our start days are saturdays

**Comment by rjc** [comment]
I'd just like to point out that Monday being the first day of the week isn't just in some regions - it's international, as in ISO - the standards organization - not just ISO 8601. If anything, that should be the default and, if not, then configurable at the very least.

**Comment by mariuszmaximus** [comment]
I'm waiting

**Comment by bendem** [comment]
Note that https://github.com/go-gitea/gitea/commit/04e97b83115e7439d43c0ede5fe2d1b50d201c52 changed the library to an updated one, I opened a new issue there: https://github.com/razorness/vue3-calendar-heatmap/issues/19

**Comment by KazzmanK** [comment]
Looks like heatmap control is abandoned one. There is a PR with fix, but still.

**Comment by Wohlstand** [comment]
> In some regions

I'll clarify you: in MANY regions, the more than a half of countries in the world uses Monday as a start of week. Also the fact: Bandgladesh begins week with Friday, and seven countries (United Arabic Emirates, Bahrein, Jibuty, Algir, Egypt, Marokko, Sudan) do begin a week with a... Saturday!

**Comment by rmatab** [comment]
> Looks like heatmap control is abandoned one. There is a PR with fix, but still. May be there is a reason to fork it and use updated version?

Last updated six years ago:

https://www.npmjs.com/package/vue-calendar-heatmap

Looks like we are waiting on someone with Vue expertise that is willing to maintain a fork or negotiate a transfer of ownership.

**Comment by bendem** [comment]
> Last updated six years ago:
> 
> https://www.npmjs.com/package/vue-calendar-heatmap

Wrong one, the library was updated, it's https://www.npmjs.com/package/vue3-calendar-heatmap, last updated 2 years ago still.

**Comment by Duviri** [comment]
a setting for this, even if its editing some line in a file somewhere and not a complete UI integration would be awesome <3

**Comment by HornetGit** [comment]
Still open after six years, so here's a workaround for anyone who can't wait.

https://github.com/HornetGit/forgejo-contribution-matrix — makes the heatmap start weeks on any day you pick (or restores the shipped layout). It overrides two frontend assets through the instance's custom public/ directory; no shipped file is modified and no binary is touched. Reverting displaces the overrides into a timestamped store rather than deleting anything.

Honest scope: Forgejo only. I tested it against Gitea 1.21.11 → 1.27.2 and every version exits 2 before touching anything — no Gitea build splits the heatmap into an activity-heatmap chunk, ≥1.26 is Vite-built with no stable index.js, and 1.27.2 dropped vue-calendar-heatmap entirely. So this isn't a fix for Gitea, just proof the frontend-override route works, and maybe a starting point if someone wants to do it properly upstream.

Verified on Forgejo 14.0.5. MIT. 277 offline assertions if you want to check it before pointing it at anything.

**Comment by wxiaoguang** [comment]
Forgejo's code base (especially frontend) is quite out-dated, so nothing to do with Gitea.

Gitea is using its own Vue component, so the problem is not "upstream" anymore, should be easy to fix.

**Comment by wxiaoguang** [comment]
-> enhance: use browser's locale to detect first (monday/sunday) for the contribution map - #38995

**Comment by wxiaoguang** [comment]
Nightly build is ready, feel free to have a try and feedback:

* https://dl.gitea.com/gitea/main-nightly/
* https://hub.docker.com/r/gitea/gitea/tags?name=main-nightly
